R/misc.R

```diff
@@ -5,17 +5,17 @@ filter_terms <- function(x, ...) {
 ## Buckets variables into discrete, mutally exclusive types
 get_types <- function(x) {
```

Comment:
First major change is happening here. The `get_types()` function used to return a single character for each class, it now returns a vector of types. This means that the results will be a list-column of types. They are captured in orders, so if something has both class `ordered` and `factor`, it will defer to the first one (which is the same behavior we had before)

``` r
tmp <- tibble::tibble(
  ord = as.ordered(letters),
  chr = letters,
  int = seq_along(letters),
  dbl = seq_along(letters) / 5
)

class(tmp$ord)
#> [1] "ordered" "factor"

recipes:::get_types(tmp)$type
#> [[1]]
#> [1] "ordered" "nominal"
#> 
#> [[2]]
#> [1] "string"    "unordered" "nominal"  
#> 
#> [[3]]
#> [1] "integer" "numeric"
#> 
#> [[4]]
#> [1] "double"  "numeric"
```